src/platform/packages/shared/kbn-discover-utils/src/data_types/traces/types.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { DataTableRecord } from '../../types';
+
+export interface TraceFields {
+  'service.name'?: string;
+  'event.outcome'?: string;
+  'transaction.name'?: string;
+  'transaction.duration.us'?: string;
+  'span.name'?: string;
+  'span.duration.us'?: string;
+  'agent.name'?: string;
+}
+
+export interface TraceDocument extends DataTableRecord {
+  flattened: {
+    '@timestamp': string;
+    'log.level'?: [string];
+    message?: [string];
+    'error.message'?: string;
+    'event.original'?: string;
+    'host.name'?: string;
+    'service.name'?: string;
+    'trace.id'?: string;
```

Comment:
Q: Why this field is optional?

Reply:
I mostly just lifted this from the `LogDocument` interface, changing it where need be. But I'm not exactly sure what the full trace document interface would look like